### PR TITLE
Modify BitField to add (+/-) near powers of two...

### PR DESCRIPTION
**Pull request by secenv**
... and surrounding numbers, as @SR4ven said in #323.

### CODE REVIEW

File: boofuzz/primitives/bit_field.py [hunk 1]
```diff
@@ -106,34 +107,59 @@ def __init__(
...
+                self.add_integer_boundaries(self.max_num // 2, 10)
+                self.add_integer_boundaries(self.max_num // 3, 10)
+                self.add_integer_boundaries(self.max_num // 4, 10)
+                self.add_integer_boundaries(self.max_num // 8, 10)
+                self.add_integer_boundaries(self.max_num // 16, 10)
+                self.add_integer_boundaries(self.max_num // 32, 10)
+                self.add_integer_boundaries(self.max_num, 10)
+
+                power_of_two = math.log(self.max_num, 2)
+                bits = width if not self.max_num else int(math.floor(power_of_two))
```
Comment (jtpereyda) [inline boofuzz/primitives/bit_field.py]: `power_of_two = math.log(self.max_num, 2)` will fail if  `self.max_num` is `None`.  Make this conditional situation more robust.

Also, for `bits = width if not self.max_num else int(math.floor(power_of_two))`, I'm not clear why the existence of max_num means you should use width.

Breaking this out into a proper if/then statement might make the whole thing look cleaner.

File: boofuzz/primitives/bit_field.py [hunk 2]
```diff
@@ -106,34 +107,59 @@ def __init__(
...
+                    # check if width is a power of 2. if it isn't, calculate and add numbers near the next
+                    # lower power of two
+                    power_of_two = math.log(width, 2)
+                    if power_of_two != math.floor(power_of_two) and power_of_two > 0:
+                        tmp = 1 << int(power_of_two)
+                        self.add_integer_boundaries(tmp, 1)
+                        if signed:
+                            self.add_integer_boundaries(-tmp, 1)
+                    bits = bits // 2
+                    tmp = 1 << bits
```
Comment (jtpereyda) [inline boofuzz/primitives/bit_field.py]: I suspect this whole chunk could be cleaned up a bit -- perhaps more helpful variable names or a modification of the layout. If you have a go I can make a second pass.

**Comment by jtpereyda** [comment]
@secenv Thank you for the PR! I left some review comments for your consideration.

**Comment by SR4ven** [comment]
Closing due to inactivity.
Feel free to reopen/comment!

**Pull request closed without merging**